From: 
Subject: Debian changes

The Debian packaging of camera-lidar-calibration is maintained in git, using a workflow
similar to the one described in dgit-maint-merge(7).
The Debian delta is represented by this one combined patch; there isn't a
patch queue that can be represented as a quilt series.

A detailed breakdown of the changes is available from their canonical
representation -- git commits in the packaging repository.
For example, to see the changes made by the Debian maintainer in the first
upload of upstream version 1.2.3, you could use:

    % git clone https://git.dgit.debian.org/camera-lidar-calibration
    % cd camera-lidar-calibration
    % git log --oneline 1.2.3..debian/1.2.3-1 -- . ':!debian'

(If you have dgit, use `dgit clone camera-lidar-calibration`, rather than plain `git clone`.)

We don't use debian/source/options single-debian-patch because it has bugs.
Therefore, NMUs etc. may nevertheless have made additional patches.

---

diff --git a/Makefile b/Makefile
index fc4d14c..61b477f 100644
--- a/Makefile
+++ b/Makefile
@@ -57,13 +57,7 @@ PODS     := $(patsubst %.py,%.py.pod,$(DIST_BIN))
 $(DIST_MAN): %.1: %.pod
 	pod2man --center="clc: camera-lidar alignment toolkit" --name=CLC --release="clc $(VERSION)" --section=1 $< $@
 
-# %.pod should NOT depend on the binary clc stuff at all: I should be able to
-# run whatever --help without building anything. But currently some tools
-# 'import clc.argparse_helpers', which runs clc/__init__.py, which loads more
-# stuff. This isn't actually used for --help, but that doesn't matter. So for
-# now I add the extra dependency. Ideally argparse_helpers and --help should not
-# need anything to be built
-%.py.pod: %.py libclc.so.$(ABI_VERSION) clc/_clc$(PY_EXT_SUFFIX)
+%.py.pod: %.py
 	$(MRBUILD_BIN)/make-pod-from-help $< > $@.tmp && cat footer.pod >> $@.tmp && mv $@.tmp $@
 EXTRA_CLEAN += $(DIST_MAN) $(PODS)
 
diff --git a/fit.py b/fit.py
index 349acc1..85df68c 100755
--- a/fit.py
+++ b/fit.py
@@ -45,9 +45,6 @@ import clc.argparse_helpers
 import re
 import os
 
-import clc
-lidar_segmentation_parameters = clc.lidar_segmentation_parameters()
-
 
 def parse_args():
 
@@ -174,22 +171,6 @@ def parse_args():
                         the intrinsics are used. The number of models given must
                         match the number of camera --topics arguments EXACTLY''')
 
-    for param,metadata in lidar_segmentation_parameters.items():
-        # Here I have a --dump that's different from the 'dump' in the
-        # segmentation parameters. I ignore the latter for now
-        if param == 'dump': continue
-
-        if metadata['pyparse'] == 'p':
-            # special case for boolean
-            parser.add_argument(f'--{param.replace("_","-")}',
-                                action  = 'store_true',
-                                help = metadata['doc'])
-        else:
-            parser.add_argument(f'--{param.replace("_","-")}',
-                                type    = type(metadata['default']),
-                                default = metadata['default'],
-                                help = metadata['doc'])
-
     args = parser.parse_args()
 
     import glob
@@ -245,6 +226,9 @@ import io
 import pickle
 import mrcal
 
+import clc
+lidar_segmentation_parameters = clc.lidar_segmentation_parameters()
+
 # ingest the lidar segmentation parameters from the arguments
 ctx = dict()
 for k in lidar_segmentation_parameters.keys():
diff --git a/lidar-segmentation.py b/lidar-segmentation.py
index 2a04556..60adcab 100755
--- a/lidar-segmentation.py
+++ b/lidar-segmentation.py
@@ -20,9 +20,6 @@ import argparse
 import re
 import os
 
-import clc
-lidar_segmentation_parameters = clc.lidar_segmentation_parameters()
-
 def parse_args():
 
     parser = \
@@ -55,18 +52,6 @@ def parse_args():
                         away before segmenting the board. This is optional. The
                         BAG may be included in BAG-CONTEXT''')
 
-    for param,metadata in lidar_segmentation_parameters.items():
-        if metadata['pyparse'] == 'p':
-            # special case for boolean
-            parser.add_argument(f'--{param.replace("_","-")}',
-                                action  = 'store_true',
-                                help = metadata['doc'])
-        else:
-            parser.add_argument(f'--{param.replace("_","-")}',
-                                type    = type(metadata['default']),
-                                default = metadata['default'],
-                                help = metadata['doc'])
-
     args = parser.parse_args()
 
     args.bag_context = getattr(args, 'bag-context')
@@ -80,6 +65,9 @@ args = parse_args()
 import numpy as np
 import numpysane as nps
 
+import clc
+lidar_segmentation_parameters = clc.lidar_segmentation_parameters()
+
 # ingest the lidar segmentation parameters from the arguments
 ctx = dict()
 for k in lidar_segmentation_parameters.keys():
